src/model/platform-validation/validate-windows.ts

```diff
@@ -4,9 +4,9 @@ import { BuildParameters } from '..';
 class ValidateWindows {
   public static validate(buildParameters: BuildParameters) {
     ValidateWindows.validateWindowsPlatformRequirements(buildParameters.targetPlatform);
-    if (!(process.env.UNITY_EMAIL && process.env.UNITY_PASSWORD)) {
-      throw new Error(`Unity email and password must be set for Windows based builds to
-                       authenticate the license. Make sure to set them inside UNITY_EMAIL
+    if (!((process.env.UNITY_EMAIL && process.env.UNITY_PASSWORD) || buildParameters.unityLicensingServer)) {
```

Comment:
Perhaps it would be helpful to simplify the if statement by adding something like

```ts
const { unityLicensingServer } = buildParameters
const hasValidLicensingCredentials = process.env.UNITY_EMAIL && process.env.UNITY_PASSWORD
const hasValidLicensingStrategy = validLicensingCredentials || unityLicensingServer 

if (!hasValidLicensingStrategy) { /* stuff */ }
```

Reply:
Agreed, I've added these variables for clarification.